cranelift/codegen/src/egraph/stores.rs

```diff
@@ -97,6 +97,32 @@ pub enum MemoryState {
     AfterInst(Inst),
 }
 
+/// Memory state index, packed into a u32.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct PackedMemoryState(u32);
+
+impl From<MemoryState> for PackedMemoryState {
+    fn from(state: MemoryState) -> Self {
+        match state {
+            MemoryState::Entry => Self(0),
+            MemoryState::Store(i) => Self(1 | (i.index() as u32) << 2),
+            MemoryState::BeforeInst(i) => Self(2 | (i.index() as u32) << 2),
+            MemoryState::AfterInst(i) => Self(3 | (i.index() as u32) << 2),
+        }
+    }
+}
+
+impl PackedMemoryState {
+    /// Does this memory state refer to a specific store instruction?
+    pub fn as_store(&self) -> Option<Inst> {
+        if self.0 & 3 == 1 {
```

Comment:
Shouldn't this be `self.0 & 3 > 0`?

Reply:
It's an unpack method just for the tag=1 case so I think this is correct?

Reply:
Ah, ok... It seemed like it could have been for all three 👍